src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,133 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] array;
+    private int size;
+
+    public ArrayList() {
+        array = new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        if (capacity > 0) {
+            array = new Object[capacity];
+        } else {
+            throw new IllegalArgumentException("Capacity must be greater than zero.");
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= array.length) {
+            grow();
+        }
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Non existent "
+                    + "position: " + index);
+        }
+        if (size >= array.length) {
+            grow();
+        }
+        Object[] result = new Object[++size];
```

Comment:
Creating a new array every time an element is added is inefficient. Consider shifting elements in the existing array instead.

Reply:
I'm forbidden from using utility methods, so i will make my own implementation in next commit

Reply:
you have arrayCopy method for this